ast/src/analyzed/display.rs

```diff
@@ -337,12 +337,63 @@ impl<T: Display> Display for LookupIdentity<T> {
     }
 }
 
+impl<T: Display> Display for PhantomLookupIdentity<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        write!(
+            f,
+            "Constr::PhantomLookup(({}, {}), [{}], {});",
+            self.left
```

Comment:
Maybe write helper functions for these sub-formatters?

Reply:
I extracted formatting the selector to a function